tests/kernel/io_uring.pm

```diff
@@ -0,0 +1,140 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Executes liburing testing suite
+# Maintainer: Andrea Cervesato <andrea.cervesato@suse.com>
+# More documentation is at the bottom
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+
+sub run {
+    my $self = shift;
+
+    select_serial_terminal;
+
+    my $repository = get_var('LIBURING_REPO', 'https://github.com/axboe/liburing.git');
+    my $timeout = get_var('LIBURING_TIMEOUT', 1800);
+    my $version = get_var('LIBURING_VERSION', '');
+    my $exclude = get_var('LIBURING_EXCLUDE', '');
+    my $issues = get_var('LIBURING_KNOWNISSUES', '');
+    my @lines;
+
+    # select latest liburing version which is supported by the system
+    if ($version eq '') {
+        my $pkginfo = script_output("zypper --no-refresh info liburing2");
+
+        foreach my $line (split /\n/, $pkginfo) {
+            if ($line =~ "Version") {
+                my @match = split /:/, $line;
+                if ($match[1] =~ '(?P<version>2.\d+)-.*') {
+                    $version = "liburing-$+{version}";
+                    last;
+                }
+            }
+        }
+    }
+
+    # install dependences
+    zypper_call("in -t pattern devel_basis git-core");
+
+    # download and compile tests
+    assert_script_run("git clone $repository");
```

Comment:
`git clone --depth 1 --no-single-branch ` will be better

Reply:
We need the whole tree to provide checkout support and select different versions provided by LIBURING_VERSION variable, so unfortunately we can't use ``depth 1`` but maybe we can use ``--single-branch`` to download ``master`` only?

Reply:
`--no-single-branch` will make sure that all branches are get (i just verified it)